src/__test__/utils/plotSpecs/generateViolinSpec.test.js

```diff
@@ -65,19 +66,10 @@ describe('generateData', () => {
     const groupingId = 'louvain';
     const plotData = generateData(cellSets, mockGeneExpression, groupingId, 'sample/sample-1');
     const expectedCells = [
-      { group: 'cluster-a', y: 0, x: MOCK_RANDOM },
-      { group: 'cluster-a', y: 1, x: MOCK_RANDOM },
-      { group: 'cluster-b', y: 2, x: MOCK_RANDOM },
-      { group: 'cluster-b', y: 3 },
-      { group: 'cluster-c', y: 4 },
-      { group: 'cluster-c', y: 5 },
+      { group: 'cluster-a', y: 0, x: MOCK_EXPECTED_RANDOM },
+      { group: 'cluster-a', y: 1, x: MOCK_EXPECTED_RANDOM },
+      { group: 'cluster-b', y: 2, x: MOCK_EXPECTED_RANDOM },
     ];
     expect(plotData.cells).toEqual(expectedCells);
   });
-
-  it('generates the no x value when null is specified', () => {
```

Comment:
Removed this test because I can't see in which acceptable situation `null` could be passed to violin plot, initial state for this config is "All" and after that it can be set to a particular cell set id or a to "All"